src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -303,13 +306,20 @@ public List<JobModel> enumerateJobs(
       filter.addFilterFlightClass(FlightFilterOp.EQUAL, className);
     }
 
+    HashSet<String> jobIdSet = new HashSet<>(ListUtils.emptyIfNull(jobIds));
     List<FlightState> flightStateList;
     boolean canListAnyJob = checkUserCanListAnyJob(userReq);
     try {
       if (canListAnyJob) {
         flightStateList = stairway.getFlights(offset, limit, filter);
+        if (!jobIdSet.isEmpty()) {
+          flightStateList =
+              flightStateList.stream()
+                  .filter(flightState -> jobIdSet.contains(flightState.getFlightId()))
```

Comment:
Filtering after calling stairway means that a user could request `10` items and get back `0` items. If you want to fix that problem, it might be worth generalizing the code in `getAccessibleFlights()` which handles this issue by making multiple calls to `stairway.getFlights()` until it's collected enough flights.

Reply:
You're totally right 🤦‍♀️ Thanks for catching that!